### PR TITLE
feat: error translator support

### PR DESCRIPTION
**Pull request by saeidee**
- [x] Do only one thing
- [x] Non breaking API changes
- [x] Tested

### What did this pull request do?

This PR adds support for translating Postgres errors like unique key violation to native gorm errors (ErrDuplicatedKey)

Related PR in gorm repo: https://github.com/go-gorm/gorm/pull/6004

### CODE REVIEW

File: error_translator.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+type ErrMessage struct {
+	Code     string `json:"Code"`
+	Severity string `json:"Severity"`
+	Message  string `json:"Message"`
+}
+
+func (dialector Dialector) Translate(err error) error {
+	parsedErr, marshalErr := json.Marshal(err)
```
Comment (jinzhu) [inline error_translator.go:19]: Maybe better to assert the error type for performance? e.g:

`pgErr, ok := err.(*pgconn.PgError)`

And use JSON marshal/unmarshal only when failing to assert

Comment (saeidee) [reply error_translator.go:19]: Yes it is a good idea, I will update the PR

Comment (jinzhu) [reply error_translator.go]: yes, but it might be broken if using a different pg driver... but anyway, both are fine for me.

Comment (Naist4869) [reply error_translator.go]:
```go
		var p *pgconn.PgError
		if errors.As(err, &p) && p.Code == "23505" {
			if strings.Contains(p.Detail, "email") {
				return nil, errno.DBError(errno.ErrEmailExist, "注册商户失败")
			}
			if strings.Contains(p.Detail, "phone") {
				return nil, errno.DBError(errno.ErrMerchantPhoneExist, "注册商户失败")
			}
		}
```
How should I tell which key is conflicting in the new version ?

Comment (Naist4869) [reply error_translator.go]: In fact I think this has been breaking API, The original code cannot be reached after updating the new version,Resulting in users not knowing whether the mailbox is duplicated or the phone number is duplicated,So I use v1.4.8

Comment (saeidee) [reply error_translator.go]: We can enhance the error to have more detail, @jinzhu what do you think?

Comment (rverdile) [reply error_translator.go]: +1 @Naist4869. Previously, I could parse pgConn.pgError for the constraint name and handle each constraint violation differently.

Comment (jinzhu) [reply error_translator.go]: Yes, agree this is a breaking API, the good news we still haven't released a new version for GORM, so these changes won't affected users unless they are using the latest master.

Maybe we need to add a configuration `TranslateError` to GORM Config, https://github.com/go-gorm/gorm/blob/master/gorm.go#L20, by default we don't translate the error.

@saeidee do you feel good to make the change?

Comment (saeidee) [reply error_translator.go]: Yes, I can implement the changes

Comment (saeidee) [reply error_translator.go]: Here is the PR https://github.com/go-gorm/gorm/pull/6178

**Pull request merged**